tests/test_models/test_data_preprocessors/test_data_preprocessor.py

```diff
@@ -95,3 +95,27 @@ def test_forward(self):
         for data_sample, expected_shape in zip(batch_data_samples, [(10, 15),
                                                                     (10, 25)]):
             self.assertEqual(data_sample.pad_shape, expected_shape)
+
+        # test cylindrical voxelization
+        point_cloud_range = [0, -180, -4, 50, 180, 2]
+        grid_size = [480, 360, 32]
+        voxel_layer = dict(
+            grid_size=grid_size,
+            point_cloud_range=point_cloud_range,
+            max_num_points=-1,
+            max_voxels=-1)
+        processor = Det3DDataPreprocessor(
+            voxel=True, voxel_type='cylindrical',
+            voxel_layer=voxel_layer).cuda()
+        num_points = 10000
+        xy = torch.rand(num_points) * 100 - 50
+        z = torch.rand(num_points) * 6 - 4
+        ref = torch.rand(num_points)
+        points = torch.stack([xy, z, ref], dim=-1).unsqueeze(0)
```

Comment:
We should also generate some points out of `point_cloud_range` and check their voxel coors.

Reply:
Expanded the range of random values. I have tested it with semantickitti samples additionally, it can handle the out range points.